packages/amplify-graphql-transformer-core/package.json

```diff
@@ -41,7 +41,8 @@
     "vm2": "^3.9.8"
   },
   "peerDependencies": {
-    "amplify-prompts": "^2.0.1"
+    "amplify-prompts": "^2.0.1",
+    "amplify-cli-core": "^4.0.0-cdkv2.2"
```

Comment:
I think `core` depends on this package. Does this introduce a circular dep here?

Reply:
yup. https://github.com/aws-amplify/amplify-cli/blob/d3dd3e8ec5f77fde60068e742a1141bf90dce7eb/packages/amplify-cli-core/package.json#L27 .

Looks like "error handling" should go out from `amplify-cli-core` into separate package.

Meanwhile we may want to hack some error/error type that bubbles up from transformer core to category-api and is translated into AmplifyError there.
Or just make sure that any error propagates and stops deployment (even if that means obscure message for a while until proper solution is in place).

Reply:
qq. out of curiosity. it doesn't blow up here because `amplify-cli-core` points to older `transformer-core` that doesn't form a cycle yet?

Reply:
i've fixed this and removed the circular dependency